.github/workflows/apidoc.yml

```diff
@@ -0,0 +1,155 @@
+name: Update API docs
+
+on:
+  # Trigger on push to branches `master` and `3.5`.
+  push:
+    branches: [ master, 3.5 ]
+  # Trigger on pull request events for PRs that have `master` as their target branch
+  pull_request:
+    branches: [ master ]
+
+  # Allow running the workflow manually
+  workflow_dispatch:
+
+defaults:
+  run:
+    # GitHub Actions adds `errexit` and `pipefail` by default, but we add `xtrace`
+    # to improve debugging some of the longer scripts.
+    shell: /bin/bash -o errexit -o pipefail -o xtrace {0}
+
+jobs:
+  main:
+    runs-on: ubuntu-20.04
```

Comment:
About a week ago, I read "`ubuntu-latest` still points to 18.04 but will change to 20.04 soon", which means that this change happens transparently to anyone who doesn't explicitly check it.

At first, I just used it to force 20.04, but given the above, I still think it's better to pin the version. I already had issues between 16.04, 18.04 and 20.04, so it's likely that issues will also arise with whatever comes after 20.04.
If we were to use `ubuntu-latest`, the version switch would happen months after Canonical's release. And I doubt that at that point anyone would think to check the OS version for weird issues like "why does this unrelated test run into a timeout now?".

Reply:
Yeap, I also noticed they where in a transition status for the label `ubuntu-latest` from 18.04 to 20.04. That's why I thought it would be better to use this label. We would have Ubuntu updates effortless.

You raised a good point, tho. I didn't expect Ubuntu updates to be source of issues... Well, I think it was actually expectable in a way.

Now I feel worried about how long GitHub is supposed to support a platform version and how often we'll need to rework/check workflows. Given Ubuntu 16.04 and 20.04 are LTS (and 16 is still an available platform), I guess Ubuntu 20 will stay for the while.

Reply:
Ubuntu LTS lasts five years, and given that 16.04 (which runs out this April) is still available in GitHub Actions, it seems that GH goes with a similar life cycle.

Of course, GH has no obligations to mirror Cannonical's LTS schedule. But given that they already limit Linux VMs to just Ubuntu, it seems likely that they would support the full LTS period.